### DISCUSSION TITLE
인자가 많은 메서드는 왜 나쁠까?

### DISCUSSION THREAD

**Discussion by seunghye218** [2024-woowacourse-study/level-interview#74] [Clean Code]
.

**Comment by alstn113** [comment]
인자가 많은 메서드는 여러 가지 이유로 나쁠 수 있습니다.

1. 인자가 많을 수록 메서드의 복잡성이 증가합니다. 각 인자의 역할과 의미를 파악하는 것이 어려워지며, 메서드의 동작을 이해하고 유지보수하는데 어려움을 겪을 수 있습니다.
2. 많은 인자를 가진 메서드는 사용자가 실수를 할 가능성이 높아집니다. 인자의 순서를 혼동하거나 필요한 값을 잘못 전달할 수 있습니다. 
3. 메서드가 많은 인자를 가지면 호출하는 쪽에서 해당 인자를 제공해야 합니다. 이는 호출하는 코드가 메서드의 내부 구현에 강하게 의존하게 되는 것을 의미합니다.

따라서 가능하면 인자의 수를 줄이고, 그룹화하여 관련된 데이터를 객체로 전달하는 방법을 고려하는 것이 좋습니다. 또한, 필요한 경우 메서드를 여러 개로 분리하여 인자의 수를 줄이고 가독성을 향상시킬 수 있습니다.

**Comment by GIVEN53** [comment]
### 장점
1. 모든 데이터를 한 번에 전달
    - 메서드 내에서 데이터를 찾거나 계산하는 추가적인 작업을 줄일 수 있다.
2. 명확한 종속성
    - 모든 종속성이 명시적으로 나열되어 있기 때문에 코드를 읽는 사람이 메서드가 어떤 종속성을 가지고 있는지 쉽게 이해할 수 있다.

### 단점
1. 복잡성 증가
    -  호출자가 올바른 인자를 전달하는지 확인해야 한다.
2. 가독성 저하
    - 호출 코드를 읽기 어렵게 만든다. 많은 인자가 호출 코드의 여러 줄에 걸쳐 표시될 수 있고, 코드의 의도를 파악하기 어려울 수 있다.
3. 유지보수 어려움
    - 메서드 시그니처를 변경할 때 모든 호출자를 수정해야 할 수 있으며, 새로운 인자가 추가될 때마다 기존 코드에 영향을 미칠 수 있다.
4. 디버깅 어려움
    - 각 인자가 메서드의 동작에 영향을 미치는지 추적하기 어려울 수 있다.
